user-interface/src/main/java/life/qbic/datamanager/views/projects/project/ontology/OntologyLookupComponent.java

```diff
@@ -0,0 +1,235 @@
+    add(ontologyGridSection);
+  }
+
+  private void updateShownResults(OntologySearchEvent event) {
+    searchTerm = event.searchValue();
+    ontologyGridLazyDataView.refreshAll();
+    foundResults.setText("%s results found".formatted(ontologyGridLazyDataView.getItems().count()));
+    if (!ontologyGridSection.isVisible()) {
+      ontologyGridSection.setVisible(true);
+    }
+  }
+
+  /**
+   * Resets the value within the searchfield, which in turn resets the grid via the fired
+   * {@link OntologySearchEvent}. Additionally, hides the entire section so the result span is only
+   * shown when the user is actively searching for an ontology
+   */
+  public void resetSearch() {
+    searchField.setValue("");
+    ontologyGridSection.setVisible(false);
+  }
+
+  /**
+   * Register an {@link ComponentEventListener} that will get informed with a
+   * {@link OntologySearchEvent}, as soon as a user searches for an ontology of interest.
+   *
+   * @param ontologySearchListener a listener on the search event trigger
+   */
+  public void addSearchEventListener(
+      ComponentEventListener<OntologySearchEvent> ontologySearchListener) {
+    addListener(OntologySearchEvent.class, ontologySearchListener);
+  }
+
+  private void initSearchField() {
+    searchField.setClassName("search-field");
+    searchField.setPlaceholder("Search for ontology term e.g. Whole blood");
+    searchField.setHelperText("Please provide at least %s letters to search for entries".formatted(
+        ONTOLOGY_SEARCH_LOWER_LIMIT));
+    searchField.setPrefixComponent(VaadinIcon.SEARCH.create());
+    searchField.setValueChangeMode(ValueChangeMode.LAZY);
+    searchField.addValueChangeListener(
+        event -> {
+          if (event.getValue().length() >= ONTOLOGY_SEARCH_LOWER_LIMIT) {
+            fireEvent(new OntologySearchEvent(this, event.isFromClient(), event.getValue()));
+          }
+        });
+    add(searchField);
+  }
+
+
+  /**
+   * Ontology Item
+   * <p>
+   * The ontology Item is a Div container styled similiar to the {@link Card} component, hosting the
+   * ontology information of interest provided by a {@link OntologyClassDTO}
+   */
+  private static class OntologyItem extends Div {
+
+    public OntologyItem(String label, String name, String classIri, String description,
+        String ontologyAbbreviation) {
+      addHeader(label, name);
+      addUrl(classIri);
+      addDescription(description);
+      addOrigin(ontologyAbbreviation);
+      addClassName("ontology-item");
+    }
+
+    private void addHeader(String label, String nameText) {
+      Span title = new Span(label);
+      title.addClassName("ontology-item-title");
+      Span name = new Tag(nameText);
+      name.addClassNames("primary", "clickable");
+      name.addClickListener(
+          event -> UI.getCurrent().getPage().executeJs("window.copyToClipboard($0)", nameText));
+      Span header = new Span(title, name, initCopyIcon(nameText));
```

Comment:
As far as I know we do not want the user to copy the name but the CURI?

Reply:
I changed the variable name, turns out we store the curie currently as a name field in the database and dtos, which is where this confusion is coming from 🤔